test/org/sagebionetworks/bridge/services/HealthDataServiceTest.java

```diff
@@ -123,4 +140,43 @@ public void getRecordsForUploadDateSuccess() {
         assertEquals("bar healthcode", recordList.get(1).getHealthCode());
         assertEquals("baz healthcode", recordList.get(2).getHealthCode());
     }
+
+    @Test
+    public void updateRecordSuccess() throws Exception {
+
+        // first create a mock record
+        // record
+        HealthDataRecord record = DAO.getRecordBuilder().withHealthCode(TEST_HEALTH_CODE)
+                .withSchemaId(TEST_SCHEMA_ID).withSchemaRevision(3).withStudyId(TEST_STUDY_ID).build();
+
+        // mock dao
+        HealthDataDao mockDao = mock(HealthDataDao.class);
+        when(mockDao.createOrUpdateRecord(record)).thenReturn(TEST_RECORD_ID);
+
+        HealthDataService svc = new HealthDataService();
+        svc.setHealthDataDao(mockDao);
+
+        // execute and validate
+        String retVal = svc.createOrUpdateRecord(record);
+        assertEquals(TEST_RECORD_ID, retVal);
```

Comment:
Following up with Alx offline, but this is not only fine to keep as is, but it's an important part of testing.

Part of the implementation is passing data from the DAO to the output of the server. It's important to test this code, to ensure you're not returning the wrong object.